api/napi/index.ts

```diff
@@ -196,15 +218,27 @@ interface Callback {
     setHandler(cb: any): void;
 }
 
+/**
+ * Represents an errors that can be emitted by the compiler.
```

Comment:
Please expand how they are emitted or how the user can get hold of them. It's often critical when browsing docs: When do I, as a user, encounter this type?

Isn't it in this case that this is what's thrown an exception? This calls for an example, or at least a reference to the function that throws it (and vice versa, but I commented on that earlier).